core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -1010,6 +1010,65 @@ private[spark] object Utils extends Logging {
     )
   }
 
+  /** Check whether a time-suffix was provided for the time string. */
+  private def hasTimeSuffix(str: String) : Boolean = {
+    val lower = str.toLowerCase.trim()
+    lower.endsWith("ms") || lower.endsWith("us") || lower.endsWith("s")
+  }
+
+  val timeError = "Time must be specified as seconds (s), " +
+      "milliseconds (ms), or microseconds (us) e.g. 50s, 100ms, or 250us."
+
+  /**
+   * Convert a passed time string (e.g. 50s, 100ms, or 250us) to a microsecond count for
+   * internal use. If no suffix is provided a direct conversion is attempted.
+   */
+  @throws(classOf[NumberFormatException])
+  private def timeStringToUs(str: String) : Long = {
+    try {
+      val lower = str.toLowerCase.trim()
+      if (lower.endsWith("ms")) {
+        lower.substring(0, lower.length - 2).toLong * 1000
+      } else if (lower.endsWith("us")) {
+        lower.substring(0, lower.length - 2).toLong
+      } else if (lower.endsWith("s")) {
+        lower.substring(0, lower.length - 1).toLong * 1000 * 1000
+      } else {
+        // Invalid suffix, force correct formatting
```

Comment:
Not really "invalid suffix", right? This is just handling the case where a suffix is not expected. If there is an invalid suffix, then an exception will be thrown.